### PR TITLE
open-webui: Add `environmentFile` option

### PR DESCRIPTION
**Pull request by anoadragon453**
## Description of changes

This option allows passing secrets to Open WebUI without directly exposing them in nix configuration.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — APPROVED** [review]

File: nixos/modules/services/misc/open-webui.nix [hunk 1]
```diff
@@ -83,6 +94,7 @@ in
 
       serviceConfig = {
         ExecStart = "${lib.getExe cfg.package} serve --host ${cfg.host} --port ${toString cfg.port}";
+        EnvironmentFile = lib.optional (cfg.environmentFile != null) cfg.environmentFile;
```
Comment (drupol) [inline nixos/modules/services/misc/open-webui.nix:97]: `lib.optional` will return a list. Is it OK for `the `EnvironmentFile` attribute?

Comment (anoadragon453) [reply nixos/modules/services/misc/open-webui.nix:97]: I cargo culted this line from:

https://github.com/NixOS/nixpkgs/blob/bc4272b10c3715186c05c5fc2f33f575d123ec8a/nixos/modules/services/web-apps/ocis.nix#L167

And the tests do show that this line works.

I tried to dig into why exactly it does work, but the systemd code is rather *obtuse*. My theory is that this bit handles wrapping values in lists if they aren't already:

https://github.com/NixOS/nixpkgs/blob/bc4272b10c3715186c05c5fc2f33f575d123ec8a/nixos/lib/systemd-lib.nix#L241-L247

Comment (drupol) [reply nixos/modules/services/misc/open-webui.nix:97]: Fair enough, all good. Thanks!

**Comment by drupol** [comment]
Oh but tests are failing now.

**Comment by anoadragon453** [comment]
> Oh but tests are failing now.

Oh! That was hard to tell from the GitHub Action CI. Turns out the tests were not passing!

I've fixed up the syntax errors, fingers crossed they pass. (I'd run the tests locally, but am on a train with very limited WiFi currently.)

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
I was not able to get it running locally either.

**Comment by drupol** [comment]
I wish I could merge this, do you mind having a look at the broken test you're trying to introduce?

**Comment by anoadragon453** [comment]
Apologies for the delay, I've had some time to fix the test and have tested it works locally. Should pass in CI now!

**Comment by anoadragon453** [comment]
Note that the CI failures on `*-darwin` appear unrelated to this PR's changes.

**Comment by anoadragon453** [comment]
@drupol are you able to have another look? 🙂

**Pull request merged**